pygfx/utils/colors.py

```diff
@@ -0,0 +1,150 @@
+import ctypes
+
+
+F4 = ctypes.c_float * 4
+
+
+class Color:
+    """An object to hold a color value."""
+
+    __slots__ = ["_val"]
+
+    def __init__(self, *args):
+
+        if len(args) == 0:
+            self._save_from_rgba(0, 0, 0, 0)
+        elif len(args) == 1:
+            color = args[0]
+            if isinstance(color, Color):
+                self._val = color._val
+            elif isinstance(color, float):
+                self._save_from_tuple((color,))
+            elif isinstance(color, int):
+                self._save_from_int(color)
+            elif isinstance(color, str):
+                self._save_from_str(color)
+            elif isinstance(color, (tuple, list)):
+                self._save_from_tuple(color)
+        else:
+            self._save_from_tuple(args)
+
+    def __repr__(self):
+        return "<Color {:0.2f} {:0.2f} {:0.2f} {:0.2f}>".format(*self.rgba)
+
+    def __len__(self):
+        return 4
+
+    def __getitem__(self, index):
+        return self._val[index]
+
+    def __iter__(self):
+        return self.rgba.__iter__()
+
+    def _save_from_rgba(self, r, g, b, a):
+        r = max(0.0, min(1.0, float(r)))
+        g = max(0.0, min(1.0, float(g)))
+        b = max(0.0, min(1.0, float(b)))
+        a = max(0.0, min(1.0, float(a)))
+        self._val = F4(r, g, b, a)
+
+    def _save_from_int(self, color):
+        v = color
+        a = v % 256
+        v = v >> 8
+        b = v % 256
+        v = v >> 8
+        g = v % 256
+        v = v >> 8
+        r = v % 256
+        self._save_from_rgba(r, g, b, a)
+
+    def _save_from_tuple(self, color):
+        color = tuple(float(c) for c in color)
+        if len(color) == 1:
+            self._save_from_rgba(color[0], color[0], color[0], 1)
+        elif len(color) == 2:
+            self._save_from_rgba(color[0], color[0], color[0], color[1])
+        elif len(color) == 3:
+            self._save_from_rgba(color[0], color[1], color[2], 1)
+        elif len(color) == 4:
+            self._save_from_rgba(*color)
+        else:
+            raise ValueError(f"Color tuple must have 1-4 value, got: {color}")
+
+    def _save_from_str(self, color):
+        color = color.lower()
+        if color.startswith("#"):
+            # todo: Hex RGB or RGBA
```

Comment:
are you going to implement this? I feel like it would be the most common use case, probably :)

Reply:
Yes, if we move forward with this idea :)